Extension/browser/webkit/webrequest.js

```diff
@@ -92,11 +101,34 @@ function onHeadersReceived(requestDetails) {
     webRequestService.processRequestResponse(tab, requestUrl, referrerUrl, requestType, responseHeaders);
 
     if (requestType == RequestTypes.DOCUMENT) {
-        //safebrowsing check
+        // Safebrowsing check
         filterSafebrowsing(tab, requestUrl);
+
+        /*
+         Websocket check.
+         If 'ws://' request is blocked for not existing domain - it's blocked for all domains.
+         Then we gonna limit frame sources to http to block src:'data/text' etc.
+         More details in the issue:
+         https://github.com/AdguardTeam/AdguardBrowserExtension/issues/344
+
+         WS connections are detected as "other"  by ABP
+         EasyList already contains some rules for WS connections with $other modifier
+         */
+        var websocketCheckUrl = "ws://adguardwebsocket.check/";
+        if (webRequestService.checkWebSocketRequest(tab, websocketCheckUrl, referrerUrl, RequestTypes.OTHER)) {

```

Comment:
I suppose we should remove `requestType` parameter from this method.

WebSocket connections are of type `OTHER`.